src/Compression/LZ4_decompress_faster.cpp

```diff
@@ -515,6 +511,18 @@ bool NO_INLINE decompressImpl(
 
         const unsigned token = *ip++;
         length = token >> 4;
+
+        UInt8 * copy_end;
+        size_t real_length;
+
+        /// It might be true fairly often for well-compressed columns.
+        /// ATST it may hurt performance in other cases because this condition is hard to predict (especially if the number of zeros is ~50%).
+        /// In such cases this `if` will significantly increase number of mispredicted instructions. But seems like it results in a
+        /// noticeable slowdown only for implementations with `copy_amount` > 8. Probably because they use havier instructions.
+        if constexpr (copy_amount == 8)
+            if (length == 0)
+                goto decompress_match;
```

Comment:
IMO, `goto` is not a good syntax, why use this?

Reply:
because it allows to add this branch only if constexpr condition is met